pygac/reader.py

```diff
@@ -120,8 +122,31 @@ def read(self, filename):
         Args:
             filename (str): Specifies the GAC/LAC file to be read.
         """
-        self.filename = os.path.basename(filename)
+        raise NotImplementedError
+
+    @contextmanager
+    def _open(self, filename):
+        """Open the GAC/LAC data file and yield the filehandle.
+
+        Args:
+            filename (str): Path to GAC/LAC file to open
+
+        Note:
+            This method should be called inside the Reader.read implementation
+        """
+        basename = os.path.basename(filename)
+        root, ext = os.path.splitext(basename)
+        if ext == ".gz":
+            self.filename = root
+            file_object = gzip.open(filename, mode='rb')
+        else:
+            self.filename = basename
+            file_object = open(filename, mode='rb')
         LOG.info('Reading %s', self.filename)
+        try:
+            yield file_object
+        finally:
+            file_object.close()
```

Comment:
Would you mind adding a small test for this function ? just to make sure it keeps working as you intend to in the future.

Reply:
If you add a gzip compressed file in test_data directory where your continuous integration checks run, it would be easy to add the test in test_klm.py and test_pod.py, by running `test_global` with the compressed file.

Reply:
Thinking again about the test, I think running test_global with a compressed file would be a little too expensive for this function. Frankly, I would be very surprised if the marked lines would break in future, because it is a standard python pattern (https://docs.python.org/3/library/contextlib.html) typically used for database sessions applying the rollback in case of an exception. However, if you prefer a test, I thought about creating a temporary file and a compressed equivalent (created in `TestGacReader.setUpClass` and deleted in `TestGacReader.tearDownClass`) and open them in a test where we could also check if the attribute filename is set correctly. What do you think?

Reply:
Yes, what I'm asking for is just a simple test that makes sure a file name ending with `.gz` is going trough gzip. I would even just mock the gzip lib and just make sure the yielded object it the return value of the `gzip.open` mock.